common/src/main/java/com/wynntils/models/activities/ContentBookQueries.java

```diff
@@ -245,19 +252,25 @@ private int getFilterChangeDirection(ItemStack itemStack, String targetFilter) {
         if (backward < 0) backward += filterCount;
 
         return forward < backward ? GLFW.GLFW_MOUSE_BUTTON_LEFT : GLFW.GLFW_MOUSE_BUTTON_RIGHT;
+         */
     }
 
     private ContainerContentVerification getContentBookFilterChangeVerification() {
         return (container, changes, changeType) -> {
             // Only set slot changes can be valid
-            if (changeType == ContainerContentChangeType.SET_CONTENT) return false;
-
-            // Check if the progress item changed, this is the last item to change
-            if (!changes.containsKey(PROGRESS_SLOT)) return false;
-
-            // Check if the filter item changed, and if so, if the active filter changed
-            String itemFilter = getActiveFilter(container.items().get(CHANGE_VIEW_SLOT));
-            return !Objects.equals(itemFilter, activeFilter);
+            // if (changeType == ContainerContentChangeType.SET_CONTENT) return false;
+
+            // TEMP FIX: Consider the filter change complete as soon as we get the first item.
+            //           We don't seem to get called for changes in the player's inventory, so we cannot
+            //           rely on the Filter or Progress items changing.
+            //           We also can't wait for the last item, since we don't know which item that might be
+            //           (some filter categories have less than a page of items, so we won't always get
+            //           change events for the last item slot)
+            //           Additionally, it appears SET_CONTENT is now used in addition to SET_SLOT, but
+            //           unconfirmed if this is intentional or a temporary bug.
+            if (changeType != ContainerContentChangeType.SET_CONTENT
+                    && changeType != ContainerContentChangeType.SET_SLOT) return false;
```

Comment:
What is the enum in this case? Null? I think this condition is either written wrongly by accident, or has a flaw in it's logic, or it masks a bigger issue in the query system (that being changeType being null).

Reply:
Briefly testing here, I did not encounter a `changeType` deviating from the two enum values. I assume the condition is faulty & could be removed